src/main/java/no/unit/nva/doi/fetch/MainHandler.java

```diff
@@ -0,0 +1,135 @@
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.net.URL;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static org.apache.http.HttpHeaders.CONTENT_TYPE;
+import static org.apache.http.HttpStatus.SC_BAD_GATEWAY;
+import static org.apache.http.HttpStatus.SC_BAD_REQUEST;
+import static org.apache.http.HttpStatus.SC_INTERNAL_SERVER_ERROR;
+import static org.apache.http.HttpStatus.SC_OK;
+import static org.apache.http.entity.ContentType.APPLICATION_JSON;
+import static org.zalando.problem.Status.BAD_REQUEST;
+import static org.zalando.problem.Status.INTERNAL_SERVER_ERROR;
+
+public class MainHandler implements RequestStreamHandler {
+
+    public static final String ACCESS_CONTROL_ALLOW_ORIGIN = "Access-Control-Allow-Origin";
+    public static final String ALLOWED_ORIGIN_ENV = "ALLOWED_ORIGIN";
+    public static final String API_HOST_ENV = "API_HOST";
+
+    private final transient ObjectMapper objectMapper;
+    private final transient PublicationConverter publicationConverter;
+    private final transient DoiTransformService doiTransformService;
+    private final transient DoiProxyService doiProxyService;
+    private final transient String allowedOrigin;
+    private final transient String apiHost;
+
+    public MainHandler() {
+        this(createObjectMapper(), new PublicationConverter(), new DoiTransformService(), new DoiProxyService(),
+                new Environment());
+    }
+
+    /**
+     * Constructor for MainHandler.
+     *
+     * @param objectMapper  objectMapper
+     * @param environment   environment
+     */
+    public MainHandler(ObjectMapper objectMapper, PublicationConverter publicationConverter,
+                       DoiTransformService doiTransformService, DoiProxyService doiProxyService,
+                       Environment environment) {
+        this.objectMapper = objectMapper;
+        this.publicationConverter = publicationConverter;
+        this.doiTransformService = doiTransformService;
+        this.doiProxyService = doiProxyService;
+        this.allowedOrigin = environment.get(ALLOWED_ORIGIN_ENV).orElseThrow(IllegalStateException::new);
+        this.apiHost = environment.get(API_HOST_ENV).orElseThrow(IllegalStateException::new);
+    }
+
+    @Override
+    public void handleRequest(InputStream input, OutputStream output, Context context) throws IOException {
+        RequestBody requestBody;
+        String authorization;
+        try {
+            JsonNode event = objectMapper.readTree(input);
+            authorization = Optional.ofNullable(event.at("/headers/Authorization").textValue())
+                    .orElseThrow(IllegalArgumentException::new);
+            String body = event.get("body").textValue();
+            requestBody = objectMapper.readValue(body, RequestBody.class);
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
```

Comment:
I wonder if we should create a logger output method that makes the reason for this System.out obvious? This can later be replaced in case we decide to use some other mechanism for logging